connection.go

```diff
@@ -213,11 +213,39 @@ func DialConfig(url string, config Config) (*Connection, error) {
 		config.Vhost = uri.Vhost
 	}
 
+	if config.Heartbeat == 0 {
+		config.Heartbeat = time.Duration(uri.Heartbeat) * time.Second
+	}
+
+	if config.ChannelMax == 0 {
+		config.ChannelMax = uri.ChannelMax
+	}
+
+	connectionTimeout := defaultConnectionTimeout
+	if uri.ConnectionTimeout != 0 {
+		connectionTimeout = time.Duration(uri.ConnectionTimeout) * time.Millisecond
+	}
+
+	if config.SASL == nil && uri.AuthMechanism != nil {
+		for _, identifier := range uri.AuthMechanism {
+			switch strings.ToUpper(identifier) {
+			case "PLAIN":
+				config.SASL = append(config.SASL, uri.PlainAuth())
+			case "AMQPLAIN":
+				config.SASL = append(config.SASL, uri.AMQPlainAuth())
+			case "EXTERNAL":
+				config.SASL = append(config.SASL, &ExternalAuth{})
+			default:
+				return nil, fmt.Errorf("unsupported auth_mechanism: %v", identifier)
+			}
+		}
+	}
```

Comment:
This needs a small tweak. A bit earlier in these lines:

https://github.com/rabbitmq/amqp091-go/blob/2e14735b182ba3adf5eeb9f457933af36d2835f6/connection.go#L208-L210

We check if SASL is `nil`, and set a plain auth mechanism. Therefore, the condition `if config.SASL == nil && uri.AuthMechanism != nil` will always evaluate to `false` (because we had already set SASL to something). This if conditional needs to happen before the first `config.SASL == nil` check.

Reply:
Moved this if to that if, should be good.